### ISSUE TITLE
Feature request: Columns should be searchable via the API, even if they were disabled during DataTable initialization

### ISSUE THREAD

**Issue by MisterVector** [DataTables/DataTablesSrc#210]
Currently, attempting to search a column via the API when searching on that column has been disabled does not return anything. I'm posting this feature request because I think that some columns may not be appropriate to search on via the search field, however through one or more filters after a search, such a column ought to be searched on. I'll try to clarify with a proof of concept.

To play around with this POC yourself, I have provided the following gists:

eligibleEmployees.html: https://gist.github.com/MisterVector/6cbbb74b74d5af7556472ec8681a5b64
eligibleEmployees.js: https://gist.github.com/MisterVector/d8b7a6c3f7b02bfce2d268c86a720ea8

The POC starts off with this table:

![image](https://user-images.githubusercontent.com/437096/172299720-58d9d75a-1de5-4da4-8c27-18d380ab1c61.png)

Scenario 1 below illustrates my point that some columns aren't appropriate to search on:

![image](https://user-images.githubusercontent.com/437096/172300055-873eb0c1-69c8-4006-a541-e72c6e80a29a.png)

Scenario 2 shows the "Show only employees eligible for promotion" option checked, which shows only the employees eligible for promotion:

![image](https://user-images.githubusercontent.com/437096/172308228-778c0b2b-7f64-4ed8-9378-a8cd0ca83777.png)

With the "Disable third column for search?" item checked, scenario 3 shows that performing the same search will not show any results, as that column has been disabled:

![image](https://user-images.githubusercontent.com/437096/172307478-58ed7806-b36d-4c2c-b92d-44a5301f6cce.png)

For scenario 4, both checkboxes have been checked, which disables the third column for search and performs a search on that column using the API, which turns up no results:

![image](https://user-images.githubusercontent.com/437096/172308510-9eca62fb-782a-4857-ba84-0da0f247dec4.png)

I'm suggesting that scenario 4 should show the intended results, regardless if the searching capabilities of that column are disabled or not.

This probably makes more sense if the column was hidden. You would disable it at initialization to make sure the hidden column is not selected when searching but then perform a search on that column using the API.

**Comment by AllanJard** [comment]
Yes, rather than just being a boolean option, it might be nice to allow it to be `searchable: 'api'`.

It isn't something I'm going to be adding soon though I'm afraid. This is the first request I can recall for this and it isn't something I've required myself yet. Will leave open for future thought.

**Comment by MisterVector** [comment]
I brought this up because I spent several hours one day at work trying to figure out what was wrong with my code, only to realize towards the end (after going through 2 other developers) that I had set one of the columns to disabled, but also that column was being searched via the API.

**Comment by MisterVector** [comment]
@AllanJard I've been working on a PR to address this issue. However, I noticed something interesting and would like your input.

I noticed that sometime back there was a change to the core.filter.js file adding an oInput.return argument when calling the _fnFilter() function:

![image](https://user-images.githubusercontent.com/437096/226192139-647ac881-b463-4ecb-950b-3d0d1cf64ba5.png)

However there is no such parameter in the _fnFilter() function. Here is its current signature:

![image](https://user-images.githubusercontent.com/437096/226191991-9725e2c9-eedc-4add-b927-07efa2612243.png)

Would you like me to remove the argument oInput.return in this PR or make a separate PR removing it? What do you see is the best path forward?

**Comment by AllanJard** [comment]
Further discussion on this was [in this PR](https://github.com/DataTables/DataTablesSrc/pull/377). Going to close here as now addressed.
